src/hooks/useHasAppUpdated/useHasAppUpdated.ts

```diff
@@ -8,39 +8,45 @@ const APP_UPDATE_CHECK_INTERVAL = 1000 * 60
 export const useHasAppUpdated = () => {
   const [hasUpdated, setHasUpdated] = useState(false)
   const [initialManifestMainJs, setManifestMainJs] = useState(null)
-  // asset-manifest tells us the latest minified built files
-  const url = '/asset-manifest.json'
+  const [initialEnvMainJs, setEnvMainJs] = useState(null)
+
+  const fetchData = async (url: string) => {
+    try {
+      const { data } = await axios.get(url)
+      return data
+    } catch (e) {
+      return console.error(`useHasAppUpdated: error fetching data from URL: ${url}`, e)
+    }
+  }
+
+  // 'asset-manifest.json' keeps track of latest minified built files.
+  // interpolated with a dummy query param to bypass the browser cache.
+  const assetManifestUrl = `/asset-manifest.json?${new Date().valueOf()}`
   const storeMainManifestJs = async () => {
-    // dummy query param bypasses browser cache
-    const { data } = await axios.get(`${url}?${new Date().valueOf()}`)
-    setManifestMainJs(data)
+    const manifestMainJs = await fetchData(assetManifestUrl)
+    setManifestMainJs(manifestMainJs)
   }
   useMemo(storeMainManifestJs, [])
```

Comment:
```suggestion
  useCallback(storeMainManifestJs, [assetManifestUrl])
```

Reply:
I think passing `[assetManifestUrl']` as a dependency to the `useCallback` hook makes the hook run only once on the component mount since `assetManifestUrl` is immutable and is guaranteed that it would not change at any time. 
Also, I thought we could store the value, then noticed we can have much benefit storing the value